src/build_helper/lib.rs

```diff
@@ -148,6 +148,41 @@ pub fn up_to_date(src: &Path, dst: &Path) -> bool {
     }
 }
 
+pub struct NativeLibBoilerplate {
+    pub skip_build: bool,
+    pub src_dir: PathBuf,
+    pub out_dir: PathBuf,
+    pub timestamp: PathBuf,
+}
+
+pub fn native_lib_boilerplate(src_name: &str,
```

Comment:
Could you add some comments here as well as to how this function is expected to be called? It looks like there's a standard rigamarole of checking `skip_build` and then creating the timestamp at the end, and it'd be good to enusre that's all documented.